### PR TITLE
[DOJOT-80] Feature: Making Dashboard Stepper Navigable

### PR DESCRIPTION
**Pull request by LeonamMoura**
https://dojot.atlassian.net/browse/DOJOT-80

### CODE REVIEW

File: MFE/dashboard/src/view/widget/wizard/wizard.jsx [hunk 1]
```diff
@@ -89,7 +139,7 @@ const Wizard = ({ initialValues, ...props }) => {
       <div className={classes.root}>
         <Stepper activeStep={page} orientation='vertical' connector={<StepLine />}>
           {steps.map(({ label, key }) => (
```
Comment (LuanEdCosta) [inline MFE/dashboard/src/view/widget/wizard/wizard.jsx]: What do you think of using the `index` of this map to navigate through steps. This way you don't need the `stepperNavigator` function.

File: MFE/dashboard/src/view/widget/wizard/wizard.jsx [hunk 2]
```diff
@@ -52,7 +74,35 @@ const Wizard = ({ initialValues, ...props }) => {
   const handleFormSubmit = value => {
     const isLastPage = page === React.Children.count(children) - 1;
     if (isLastPage) {
-      return onSubmit(value);
+      console.log(value);
```
Comment (LuanEdCosta) [inline MFE/dashboard/src/view/widget/wizard/wizard.jsx]: Remove this console.log

File: MFE/dashboard/src/view/widget/wizard/wizard.jsx [hunk 3]
```diff
@@ -52,7 +74,35 @@ const Wizard = ({ initialValues, ...props }) => {
...
+
+      if (value.general.name.length < 5) {
```
Comment (LuanEdCosta) [inline MFE/dashboard/src/view/widget/wizard/wizard.jsx]: What are these lines about? Why adding them in this PR?

**Review by LuanEdCosta — CHANGES_REQUESTED** [review]

File: MFE/dashboard/src/view/widget/wizard/wizard.jsx [hunk 4]
```diff
@@ -40,6 +40,28 @@ const Wizard = ({ initialValues, ...props }) => {
     setValues(value);
   };
 
+  const stepperNavigator = label => {
```
Comment (luisfelipefurlan) [inline MFE/dashboard/src/view/widget/wizard/wizard.jsx]: You cannot define the steps inside the wizard component, each widget can have its own steps with an amount defined by the widget itself.

File: MFE/dashboard/src/view/widget/wizard/wizard.jsx [hunk 5]
```diff
@@ -52,7 +74,35 @@ const Wizard = ({ initialValues, ...props }) => {
   const handleFormSubmit = value => {
     const isLastPage = page === React.Children.count(children) - 1;
     if (isLastPage) {
-      return onSubmit(value);
+      console.log(value);
+
+      if (value.general.name.length < 5) {
+        dispatchEvent(EVENT.GLOBAL_TOAST, {
```
Comment (luisfelipefurlan) [inline MFE/dashboard/src/view/widget/wizard/wizard.jsx]: Validation has to be done by the widget and not within the wizard. Each widget has to define its rules.

**Review by luisfelipefurlan — CHANGES_REQUESTED** [review]

**Review by LuanEdCosta — APPROVED** [review]

**Pull request closed without merging**